http-common/src/proxy.rs

```diff
@@ -0,0 +1,196 @@
+                }
+                Some((key, certs)) => {
+                    // DEVNOTE: SslConnectionBuilder::build() consumes the builder. So, we need
+                    //          to create two copies of it.
+                    let mut tls_connector =
+                        openssl::ssl::SslConnector::builder(openssl::ssl::SslMethod::tls())?;
+                    let mut proxy_tls_connector =
+                        openssl::ssl::SslConnector::builder(openssl::ssl::SslMethod::tls())?;
+
+                    let mut device_id_certs =
+                        openssl::x509::X509::stack_from_pem(&certs)?.into_iter();
+                    let client_cert = device_id_certs.next().ok_or_else(|| {
+                        io::Error::new(io::ErrorKind::Other, "device identity cert not found")
+                    })?;
+
+                    tls_connector.set_certificate(&client_cert)?;
+                    proxy_tls_connector.set_certificate(&client_cert)?;
+
+                    for cert in device_id_certs {
+                        tls_connector.add_extra_chain_cert(cert.clone())?;
+                        proxy_tls_connector.add_extra_chain_cert(cert.clone())?;
+                    }
+
+                    tls_connector.set_private_key(&key)?;
+                    proxy_tls_connector.set_private_key(&key)?;
+
+                    let mut http_connector = hyper::client::HttpConnector::new();
+                    http_connector.enforce_http(false);
+                    let tls_connector = hyper_openssl::HttpsConnector::with_connector(
+                        http_connector,
+                        tls_connector,
+                    )?;
+                    let mut proxy_connector =
+                        hyper_proxy::ProxyConnector::from_proxy(tls_connector, proxy)?;
+                    proxy_connector.set_tls(Some(proxy_tls_connector.build()));
+                    proxy_connector
+                }
+            };
+            Ok(MaybeProxyConnector::Proxy(proxy_connector))
+        } else {
+            let https_connector = hyper_openssl::HttpsConnector::new()?;
+            Ok(MaybeProxyConnector::NoProxy(https_connector))
+        }
+    }
+}
+
+#[cfg(feature = "tokio1")]
+impl<C> hyper::service::Service<http::uri::Uri> for MaybeProxyConnector<C>
+where
+    C: hyper::service::Service<http::uri::Uri> + Send + Unpin + 'static,
+    C::Response:
+        AsyncRead + AsyncWrite + hyper::client::connect::Connection + Send + Unpin + 'static,
+    C::Future: Send + 'static,
+    C::Error: Into<Box<dyn std::error::Error + Sync + Send>>,
+{
+    type Response = MaybeProxyStream<C::Response>;
+    type Error = io::Error;
+    type Future = Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send>>;
+
+    fn poll_ready(
+        &mut self,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        std::task::Poll::Ready(Ok(()))
```

Comment:
nit: you have `use std::task::Poll` in scope, so no need to use the fully qualified path here (same with `Context`)

Reply:
Done.